Tick cleanup

pixelherodev opened the pull request:
Most importantly, adds conversion functions that let us go from ticks <-> unix timestamps without passing through Go time.Time first. Removes allocations and computations.

docmerlin commented on sttp/ticks/Ticks.go:
```diff
@@ -28,99 +30,43 @@ import (
-	return time.Unix(0, int64((ticks-UnixBaseOffset)&ValueMask)*100).UTC()
+// Converts a unix nanoseconds timestamp into a Ticks value.
+func FromUnixNs(ns uint64) Ticks {
+	return Ticks(ns / 100) + UnixBaseOffset
```
Instead of replacing the ToTime function, we should just add the FromUnixNs function.  That way we don't suggest breaking changes to a public API.

Also, I suggest naming `FromUnixNs` as `FromUnixNano` just to be consistent with `time.UnixNano()` method in the golang std lib.

Commit:
subscriber: switch from ToTime(ticks) to ticks.ToTime() API

Using the member function is more idiomatic; having two identical APIs
is redundant and ideally we should drop the secondary one.

Commit:
ticks: Drop redundant explanation

The explanation is not sufficient for those not versed in time standards
and not necessary for those who are; the comment does not really help.

Commit:
ticks: add ticks.FromUnixNs() conversion function

Bypasses using time.Time as an intermediary, reducing a source of
per-point overhead.

Commit:
ticks: switch FromTime() to use FromUnixNs()

FromTime() conversion already works by converting to unix nanoseconds
first, and then to Ticks; just reuse the FromUnixNs() function instead
of implementing twice.

Commit:
ticks: have Ticks.ToTime() implement logic directly

The member function is more idiomatic; it should implement logic
directly instead of relying upon a secondary implementation.

Ideally, the secondary API should be dropped.

Commit:
ticks: add Ticks.ToUnixNs() function

Similar to FromUnixNs(), bypassing intermediate time.Time conversion
reduces necessary computation and stack usage, and potentially
allocations?

I've verified that this improves (reduces) CPU utilization, with user
code switched to use this API.

Commit:
ticks: document that leap seconds are discarded upon format conversion

Converting from ticks to time.Time or nanoseconds is an inherently lossy
operation!

Commit:
ticks: expose leap second APIs as part of the Ticks interface

This should ideally replace the existing ticks.*LeapSecond* functions,
so that we don't have some functions as fn(Ticks) and others as
Ticks.fn()

Commit:
ticks: test unix timestamp conversion

Commit:
switch from ToTime(ticks) to ticks.ToTime()

Commit:
ticks: update modification history

Commit:
Fix MaxTicks comment

Commit:
fix whitespace

Commit:
fix nano usage

Commit:
ticks: rename from UnixNs to UnixNano for consistency with Time

pixelherodev commented:
https://app.shortcut.com/pingthing/story/31266/sttp-tick-reworks-expose-optimization-potential
